cmd/kubeadm/app/apis/kubeadm/v1beta1/types.go

```diff
@@ -274,6 +277,9 @@ type JoinConfiguration struct {
 
 	// FeatureGates enabled by the user.
 	FeatureGates map[string]bool `json:"featureGates,omitempty"`
+
+	// Timeouts defines maximum time intervals kubeadm should wait for completion of critical steps of the init workflow.
+	Timeouts map[KubeadmTimeout]metav1.Duration
```

Comment:
We currently have a discovery timeout `Discovery.Timeout`, should we move or link that in here as another key?